music21/musicxml/m21ToXml.py

```diff
@@ -3350,99 +3395,100 @@ def parseFlatElements(
                     self.moveForward(offsetToMoveForward)
 
                 for obj in objGroup:
-                    self.parseOneElement(obj, prePostObjectSpannersOnly=True)
+                    self.parseOneElement(obj, AppendSpanners.RELATED_ONLY)
 
         if backupAfterwards:
             # return to the beginning of the measure.
             if self.offsetInMeasure:
                 self.moveBackward(self.offsetInMeasure)
+        else:
+            # if necessary, jump to end of the measure.
+            if self.offsetInMeasure < firstPassEndOffsetInMeasure:
+                self.moveForward(firstPassEndOffsetInMeasure)
 
         self.currentVoiceId = None
 
     def parseOneElement(
         self,
         obj,
-        prePostObjectSpannersOnly: bool = False,
-        skipPrePostObjectSpanners: bool = False
+        appendSpanners: AppendSpanners = AppendSpanners.NORMAL
```

Comment:
This is the most AWESOME implementation of a refactor ever.  I'm so happy with how the logic flows here now.  Thank you thank you thank you thank you!